src/index.ts

```diff
@@ -78,48 +78,55 @@ export default class TabContainerElement extends HTMLElement {
       }
     }
   }
-}
-
-function selectTab(tabContainer: TabContainerElement, index: number) {
-  const tabs = getTabs(tabContainer)
-  const panels = Array.from(tabContainer.querySelectorAll<HTMLElement>('[role="tabpanel"]')).filter(
-    panel => panel.closest(tabContainer.tagName) === tabContainer
-  )
-
-  const selectedTab = tabs[index]
-  const selectedPanel = panels[index]
 
-  const cancelled = !tabContainer.dispatchEvent(
-    new CustomEvent('tab-container-change', {
-      bubbles: true,
-      cancelable: true,
-      detail: {relatedTarget: selectedPanel}
-    })
-  )
-  if (cancelled) return
-
-  for (const tab of tabs) {
-    tab.setAttribute('aria-selected', 'false')
-    tab.setAttribute('tabindex', '-1')
-  }
-  for (const panel of panels) {
-    panel.hidden = true
-    if (!panel.hasAttribute('tabindex') && !panel.hasAttribute('data-tab-container-no-tabstop')) {
-      panel.setAttribute('tabindex', '0')
+  selectTab(index: number): void {
+    const tabs = getTabs(this)
+    const panels = Array.from(this.querySelectorAll<HTMLElement>('[role="tabpanel"]')).filter(
+      panel => panel.closest(this.tagName) === this
+    )
+
+    /**
+     * Out of bounds index
+     */
+    if (index > tabs.length - 1) {
+      throw new Error(`Cannot select tab at index "${index}" as it exceeds the total number of tabs`)
```

Comment:
Perhaps a more succinct error message here:

```suggestion
      throw new RangeError(`Index "${index}" out of bounds`)
```

Reply:
Sure :)